### PR TITLE
[Doc] v5.1.0 Release Notes

### PR DESCRIPTION
**Pull request by furszy**
Finalized release notes for v5.1.0

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+### Consensus
+- #2062 Max block space available for shielded txes (random-zebra)
+- #2051 Bump TxVersion::SAPLING to 3 (random-zebra)
+- #2068 Budget: Fix instant killing of proposals (random-zebra)
+- #2069 Start new Testnet (v5) (random-zebra)
+- #2071 Update Spork keys for v5 (random-zebra)
+- #2084 [BUG][Masternode] Miner not paying valid finalized budget if <20 active masternodes (random-zebra)
+- #2101 Max payment count for budget proposals (random-zebra)
+- #2112 Set v5 enforcement height for mainnet (furszy)
+- #2137 Fix superblock block value (random-zebra)
```
Comment (random-zebra) [inline doc/release-notes.md:68]: All these were part of either 5.0.0 or 5.0.1

File: doc/release-notes.md [hunk 2]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+- #2243 [Protocol] Block version 9 (random-zebra)
+
+### Block and transaction handling
+- #2067 Reject outdated block version after v5 enforcement (random-zebra)
```
Comment (random-zebra) [inline doc/release-notes.md:72]: part of 5.0.0

File: doc/release-notes.md [hunk 3]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+- #2150 Revamping block validation interface interaction with wallet (furszy)
+- #2192 Support for validation interface signals in background thread (furszy)
+- #2203 Call wallet/validation notify callbacks in scheduler thread (without cs_main) (furszy)
+- #2213 Remove dead zc checks from ATMP (furszy)
+- #2226 [Refactoring] Prune invalid UTXOs from the coins cache (random-zebra)
+- #2295 [Net_Processing] Cleaner block processing workflow (furszy)
+- #2290 [Test] Fix concurrency issues in ActivateBestChain + optimization (furszy)
+- #2284 Removing unneeded block serialization and signal from ABC (furszy)
+- #2052 [Miner] Update block header's current version to v8 (Fuzzbawls)
```
Comment (random-zebra) [inline doc/release-notes.md:81]: part of 5.0.0. 
2243 supersedes it in 5.1.0

File: doc/release-notes.md [hunk 4]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+
+### P2P protocol and network code
+- #2054 Socks5 netbase back ports, less scary failures (furszy)
```
Comment (random-zebra) [inline doc/release-notes.md:84]: in 5.0.0

File: doc/release-notes.md [hunk 5]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+- #2133 Remove unused RecvLine function (furszy)
+- #2130 [Backport][Net] Do not make it trivial for inbound peers to generate log entries (furszy)
+- #2134 [BUG] Incorrect name logged for inventory messages (random-zebra)
+- #2129 [Backport] Check interruptNet during dnsseed lookups (furszy)
+- #2090 Fail instead of truncate command name in CMessageHeader (random-zebra)
+- #2118 Inv, mempool and validation improvements (furszy)
+- #2172 Improve and document SOCKS code (random-zebra)
+- #2045 [TierTwo] Remove unneeded extra logging (furszy)
```
Comment (random-zebra) [inline doc/release-notes.md:92]: in 5.0.0

File: doc/release-notes.md [hunk 6]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+
+### Policy
+- #2108 Set DEFAULT_SHIELDEDTXFEE_K to 100 (from 1000) (random-zebra)
```
Comment (random-zebra) [inline doc/release-notes.md:95]: in 5.0.0

File: doc/release-notes.md [hunk 7]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
 
 ### Wallet
+- #2046 [Sapling] Transparent inputs selection, fixing inverted order (furszy)
+- #2048 BugFix: automatic upgrade for unlocked wallet (furszy)
+- #2065 [Sapling] Dummy ECDSA sigs and zk-proofs during fee calculation loop (random-zebra)
+- #2080 [BUG][Wallet] return unconfirmed balance for all spendable coins by default (random-zebra)
+- #2109 clear StakeableCoins before initializing (PeterL73)
```
Comment (random-zebra) [inline doc/release-notes.md:102]: in 5.0.0

File: doc/release-notes.md [hunk 8]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+- #2209 Killing wallet and GUI cs_main locks (furszy)
+
+### RPC and other APIs
+- #2049 Add nullifiers to listshieldedunspent output (random-zebra)
+- #2050 redirect sendtoaddress to shieldedsendmany for shielded recipient (random-zebra)
+- #2088 Cleanup RPC help output (Fuzzbawls)
+- #2086 Use "shield" nomenclature instead of "shielded" (Fuzzbawls)
+- #2097 Fix more RPC help output (Fuzzbawls)
+- #2119 Fix inconsistencies in outputs (Fuzzbawls)
+- #2120 Update Sapling HD chain seed with sethdseed (random-zebra)
```
Comment (random-zebra) [inline doc/release-notes.md:121]: in 5.0.0

File: doc/release-notes.md [hunk 9]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+- #2141 Fix argument parsing and example for getbalance (Fuzzbawls)
+- #2147 fix getbalance transparent watch-only amount (furszy)
+- #2154 `validateaddress` remove duplicate return address value (furszy)
+- #2157 mnbudgetvote generalized, duplicated code cleaned (furszy)
+- #2177 [BUG] Fix getblockindexstats-getfeeinfo with shield txes (random-zebra)
+- #2283 `getwalletinfo`: Add last_processed_block return value (furszy)
+- #2047 [BUG][RPC] Fix check in wallet upgrade RPC (random-zebra)
```
Comment (random-zebra) [inline doc/release-notes.md:128]: in 5.0.0

File: doc/release-notes.md [hunk 10]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+- #2094 Hide shield all coins button (furszy)
+- #2096 [BugFix] Duplicate transaction records (furszy)
+- #2104 [BUG][GUI] Don't return StakingOnlyUnlocked from WalletModel::sendCoins (random-zebra)
+- #2102 [BUG][GUI] Cache unconfirmed_balance only for transparent outs (random-zebra)
+- #2111 [BUG][GUI] coin-control dialog: more accurate "after fee" + fDust for shield recipient (random-zebra)
+- #2100 Support for op_return proposal output records (furszy)
+- #2122 [BUG][GUI] Fix double conf dialog for RPC commands that expose privkeys (random-zebra)
+- #2115 Fixing tooltip balance amount cropped (furszy)
+- #2125 Update Translations from Transifex (Fuzzbawls)
+- #2128 Receive screen, fix missing "Add label" text (furszy)
```
Comment (random-zebra) [inline doc/release-notes.md:146]: Either in 5.0.0 or 5.0.1 already

File: doc/release-notes.md [hunk 11]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+- #2247 Fix double fade-in Topbar Available click animation. Issue 2216 (PicklesRcool)
+- #2237 Set fee to highest possible if input is too big (dnchk)
+- #2215 Automatically set the lowest possible Custom Fee when user provided fee is too low (MishaPozdnikin)
+
+### Tests and QA
+- #2042 Adding sapling transaction validation unit test coverage (furszy)
+- #2059 Make rpc_ban unit test pass again (furszy)
+- #2081 minor fixes for mining_pos_reorg.py (random-zebra)
+- #2116 Print error if shield diversified address is equal to the default address (furszy)
+- #2123 Attempt to fix the diversified address equal to default address test scenario (furszy)
```
Comment (random-zebra) [inline doc/release-notes.md:166]: in 5.0.0

File: doc/release-notes.md [hunk 12]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+- #2206 script standard tests back ported (furszy)
+- #2214 Fix various things -fsanitize complains about (random-zebra)
+- #2221 updating rpc_invalidateblock.py to latest upstream (furszy)
+- #2233 fix miner_tests block generation race condition (furszy)
+- #2235 address.py: fix invalid `keyhash_to_p2pkh()` (furszy)
+- #2254 sync_blocks and sync_mempool improvements (random-zebra)
+
+### Build system
+- #2041 CMake pivxd and unit tests: fixing missing MacOS framework (furszy)
+- #2087 Unify header file extensions (Fuzzbawls)
```
Comment (random-zebra) [inline doc/release-notes.md:187]: in 5.0.0

File: doc/release-notes.md [hunk 13]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+- #2060 [BUG][MN] Immediately detect forked masternodes (random-zebra)
+- #2103 [Startup] Do not block local peer port selection for Masternodes running over tor (furszy)
+- #2136 [BUG][Masternodes] Prevent broadcast of pre-5.0 stale masternode data (random-zebra)
+- #2143 [BUG] Fix deadlock in GetNextMasternodeInQueueForPayment (random-zebra)
+- #2170 Better pivx.conf Masternode data error notification (furszy)
+- #2186 Improving masternode check inputs and add process (furszy)
+
+### Miscellaneous
+- #2064 script: [bug] fix signature malleability for t inputs in sapling txes (random-zebra)
+- #2083 [Core] Prevector Optimizations (random-zebra)
```
Comment (random-zebra) [inline doc/release-notes.md:205]: in 5.0.0

File: doc/release-notes.md [hunk 14]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+- #2142 Bump master to 5.0.99 (Fuzzbawls)
+- #2152 [CMake] Fix CMake builds on macOS Big Sur (Fuzzbawls)
+- #2105 Don't compile PoW mining RPC commands by default (Fuzzbawls)
+- #2198 [GA] Enable CMake macOS Build (Fuzzbawls)
+- #2227 fixing gitian build fail with checksum mismatch (furszy)
+- #2230 [GA] Dont run the feature_dbcrash test in GA (Fuzzbawls)
+
+### Tier Two Network (Masternodes / Budget)
+- #2060 [BUG][MN] Immediately detect forked masternodes (random-zebra)
+- #2103 [Startup] Do not block local peer port selection for Masternodes running over tor (furszy)
```
Comment (random-zebra) [inline doc/release-notes.md:197]: in 5.0.0

File: doc/release-notes.md [hunk 15]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+- #2205 logging: Remove unused return value from logprintstr (random-zebra)
+- #2223 [BUG] Fix some chainstate-init-order bugs (random-zebra)
+- #2166 [BUG] Flush the chainstate after pruning invalid entries (random-zebra)
+- #2191 [BugFix] Shutdown, stop threadGroup before dump data to disk (furszy)
+- #2224 Removing `TX_ZEROCOINMINT` and `ScriptPubKeyToJSON` (furszy)
+
+### Documentation
+- #2095 Main readme re-written, updated to the current project state (furszy)
+- #2113 RPC getrawtransaction and decoderawtransaction, sapling tx data (furszy)
+- #2114 v5 Release Notes (furszy)
```
Comment (random-zebra) [inline doc/release-notes.md:222]: in 5.0.0

File: doc/release-notes.md [hunk 16]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+- #2262 add reduce-memory.md (furszy)
+
+### Refactoring
+- #2056 [Masternode] Encapsulate global and do not parse privkey redundantly everywhere (furszy)
+- #2061 Add Zcash copyright statement (random-zebra)
+- #2033 Adjust CoinControl column widths for readability (NoobieDev12)
+- #2091 [Refactor] Pass caught logic_error by reference in CreateSig (random-zebra)
+- #2093 [RPC] Drop 'ed' suffix from vShielded* in TxSaplingToJSON (random-zebra)
+- #2098 [GUI] coin control: Fixed column size in list-mode (random-zebra)
+- #2121 Save cold-staking address in contacts (random-zebra)
```
Comment (random-zebra) [inline doc/release-notes.md:235]: in 5.0.0

File: doc/release-notes.md [hunk 17]
```diff
@@ -38,46 +38,265 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+- #2175 [Refactoring] Remove g_IsSapling active flag (random-zebra)
+- #2164 Several updates to the base chain params structure (furszy)
+- #2200 Minor content change for more precise description (NoobieDev12)
+- #2181 [Refactoring] Sapling classes - initialize fields and pass parameters by reference (random-zebra)
+- #2182 Default initialization for scalar and pointer fields (random-zebra)
+- #2180 [Refactoring] Remove BaseOutPoint::GetHash (random-zebra)
+- #2204 [Refactoring] Verify MessageSigner signatures with keyIDs (random-zebra)
+- #2210 Implement CheckColdStake in TransactionSignatureChecker (random-zebra)
+- #2212 [Misc] Init fixes + DisconnectBlock cleanup + zerocoin dead code removal (furszy)
+- #2058 [Cleanup] Removing unused DisconnectBlocks and ReprocessBlocks (furszy)
```
Comment (random-zebra) [inline doc/release-notes.md:251]: in 5.0.0

**Review by random-zebra — CHANGES_REQUESTED** [review]
Looks like the script pulled a bunch of PRs that were part of the previous releases. 
Not sure if I got them all.

**Comment by furszy** [comment]
ok done, thanks for the feedback. Need to tweak this new script a bit more.
Went through all the lines one by one manually, should be fine now.

**Review by random-zebra — APPROVED** [review]
utACK 925e74f1c0d05aa75ce856e4d1e9413bd3aaab51

File: doc/release-notes.md [hunk 18]
```diff
@@ -38,46 +38,191 @@ Apple released it's last Mountain Lion update August 13, 2015, and officially en
...
+- Improved chain-state corruption recovery.
+- Improved core performance and stability.
+- A good amount of bugs fixed.
 
 #### Disable PoW mining RPC Commands
 
 A new configure flag has been introduced to allow more granular control over weather or not the PoW mining RPC commands are compiled into the wallet. By default they are not. This behavior can be overridden by passing `--enable-mining-rpc` to the `configure` script.
 
-
 *version* Change log
```
Comment (Fuzzbawls) [inline doc/release-notes.md:54]: `*version*` -> `v5.1.0`

**Review by Fuzzbawls — COMMENTED** [review]
The first two lines of the file should be removed, and instances of `*version*` replaced with the version (ie `v5.1.0`) (line 3 and the inline comment location).

Also this above the `How to Upgrade` section:
```
Recommended Update
==============

This version is an optional, but recommended, update for all users and services.
```

Otherwise good to go

File: doc/release-notes.md (hunk 18 above)
Comment (furszy) [reply doc/release-notes.md]: done

**Comment by furszy** [comment]
Updated per feedback.

**Review by random-zebra — APPROVED** [review]
re-utACK 8993dd122ba3c89688b4533a8a9bc24eca8c4a3a

**Comment by furszy** [comment]
Merging and tagging.. 🚜

**Pull request merged**